app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/GoogleCloudBuildStageForm.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import Select, { Option } from 'react-select';
+import { get } from 'lodash';
+
+import {
+  ArtifactTypePatterns,
+  IArtifact,
+  IExpectedArtifact,
+  IFormikStageConfigInjectedProps,
+  RadioButtonInput,
+  SETTINGS,
+  StageArtifactSelector,
+  StageConfigField,
+  yamlDocumentsToString,
+  YamlEditor,
+} from '@spinnaker/core';
+
+export enum buildDefinitionSources {
+  ARTIFACT = 'artifact',
+  TEXT = 'text',
+}
+
+interface IGoogleCloudBuildStageFormProps {
+  googleCloudBuildAccounts: string[];
+}
+
+interface IGoogleCloudBuildStageFormState {
+  rawBuildDefinitionYaml: string;
+}
+
+export class GoogleCloudBuildStageForm extends React.Component<
+  IGoogleCloudBuildStageFormProps & IFormikStageConfigInjectedProps,
+  IGoogleCloudBuildStageFormState
+> {
+  public constructor(props: IGoogleCloudBuildStageFormProps & IFormikStageConfigInjectedProps) {
+    super(props);
+    const stage = props.formik.values;
+    this.state = {
+      rawBuildDefinitionYaml: stage.buildDefinition ? yamlDocumentsToString([stage.buildDefinition]) : '',
+    };
+  }
+
+  private onYamlChange = (rawYaml: string, yamlDocs: any): void => {
+    this.setState({ rawBuildDefinitionYaml: rawYaml });
+    const buildDefinition = Array.isArray(yamlDocs) && yamlDocs.length > 0 ? yamlDocs[0] : null;
+    this.props.formik.setFieldValue('buildDefinition', buildDefinition);
+  };
+
+  private onAccountChange = (accountOption: Option) => {
+    this.props.formik.setFieldValue('account', accountOption.value);
+  };
+
+  private getAccountOptions = (): Array<Option<string>> => {
+    return this.props.googleCloudBuildAccounts.map(account => ({
+      label: account,
+      value: account,
+    }));
+  };
+
+  private getSourceOptions = (): Array<Option<string>> => {
+    return [
+      { value: buildDefinitionSources.TEXT, label: 'Text' },
+      { value: buildDefinitionSources.ARTIFACT, label: 'Artifact' },
+    ];
+  };
+
+  private onSourceChange = (e: any) => {
+    this.props.formik.setFieldValue('buildDefinitionSource', e.target.value);
+  };
+
+  private onExpectedArtifactSelected = (expectedArtifact: IExpectedArtifact) => {
```

Comment:
@jkschneider I tagged you on this to make sure I'm using your lovely new component as intended -- we need to post either 1) just an artifact ID if a pre-existing artifact is selected or 2) a complete artifact if defined inline, correct?